db/migrations/2022120207400000-create-leaderboard-users-already-existing-on-contract.js

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable no-prototype-builtins */
+const { QueryTypes } = require("sequelize");
+
+const { Settings } = require("../models/settings.model");
+const { LeaderBoard } = require("../models/leaderboard.model");
+
+const { Web3Connection, BountyToken, NetworkRegistry } = require("@taikai/dappkit");
+require("dotenv").config();
+
+const name = "create-leaderboard-users-already-existing-on-contract";
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    if (process.env?.SKIP_MIGRATION_SEED_LEADERBOARD?.toLowerCase() === "true")
+      return console.log("SKIPPING SEED LEADERBOARD STEP");
+
+    const settings = await queryInterface.sequelize.query(
+      "SELECT * FROM settings WHERE visibility = :public",
+      {
+        replacements: { public: "public" },
+        model: Settings,
+        mapToModel: true,
+        type: QueryTypes.SELECT,
+      }
+    );
+
+    if (!settings.length) return console.log(`${name} - Settings not found`);
+
+    const networkRegistry = settings.find(
+      ({ key }) => key === "networkRegistry"
+    );
+
+    if (!networkRegistry)
+      return console.log(`${name} - Network registry not found`);
+
+    console.log("Begin create and change users leaderboard");
+
+    let usersUpdated = 0;
+
+    const web3Connection = new Web3Connection({
+      skipWindowAssignment: true,
+      web3Host: process.env.NEXT_PUBLIC_WEB3_CONNECTION,
+    });
+
+    await web3Connection.start();
+
+    const registry = new NetworkRegistry(web3Connection, networkRegistry.value);
+    await registry.loadContract();
+
+    const _bountyToken = new BountyToken(
+      web3Connection,
+      registry.bountyToken.contractAddress
+    );
+    await _bountyToken.loadContract();
+
+    const blockNumber = await web3Connection.eth.getBlockNumber();
+
+    const TransferEvents = await _bountyToken.getTransferEvents({
```

Comment:
This has to be paginated, as the distance between start and finish can be too great for the api to handle